src/Interpreters/ExpressionActions.cpp

```diff
@@ -1018,12 +1018,12 @@ void ExpressionActionsChain::ArrayJoinStep::finalize(const NameSet & required_ou
 
     for (const auto & column : result_columns)
     {
-        if (array_join->columns.count(column.name) != 0 || required_output_.count(column.name) != 0)
+        if (!array_join->columns.contains(column.name) || required_output_.contains(column.name))
```

Comment:
```suggestion
        if (array_join->columns.contains(column.name) || required_output_.contains(column.name))
```

Reply:
That one slipped through. Fixed, thanks a lot.
Will review the change once more by myself to make sure there are no other such issues.